web_src/js/components/RepoBranchTagSelector.vue

```diff
@@ -30,6 +30,7 @@
         <div class="loading-indicator is-loading" v-if="isLoading"/>
         <div v-for="(item, index) in filteredItems" :key="item.name" class="item" :class="{selected: item.selected, active: active === index}" @click="selectItem(item)" :ref="'listItem' + index">
           {{ item.name }}
+          <div class="ui label" v-if="index===0 && mode === 'branches'">{{ textDefault }}</div>
```

Comment:
So we need to ensure the first branch is the default one?

Reply:
https://github.com/go-gitea/gitea/blob/b62c8e7765a371600a300f62da96483a1ae0c731/routers/web/repo/repo.go#L648

emmm, it seems that in sometimes there's no default branch?

Reply:
No. I mean we need more comments both the two places.